timewarrior: new port

herbygillot opened the pull request:
#### Description

New port for **[TimeWarrior](https://timewarrior.net)**, a time tracking utility from the same [author](https://gothenburgbitfactory.org) as [TaskWarrior](https://taskwarrior.org).

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
printf "%s\n" "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)" "$(xcodebuild -version|awk 'NR==1{x=$0}END{print x" "$NF}')"|tee /dev/tty|pbcopy
-->
macOS 10.15.7 19H114
Xcode 12.3 12C33

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -d install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

herbygillot committed:
timewarrior: new port

Tatsh commented:
I would suggest adding `github.livecheck.regex  {([0-9.]+)}`. I had an issue with this project where `1.4.2-pre` was thought to be newer than `1.4.2`. So we have to ignore the `-pre` releases. I filed a ticket about this but changing the livecheck `vercmp` function isn't really desired.

How is this working without libshared? The release tarball has it included?

[My Portfile](https://github.com/Tatsh/ports/blob/master/office/timewarrior/Portfile)

herbygillot commented:
> I would suggest adding `github.livecheck.regex {([0-9.]+)}`. I had an issue with this project where `1.4.2-pre` was thought to be newer than `1.4.2`. So we have to ignore the `-pre` releases. I filed a ticket about this but changing the livecheck `vercmp` function isn't really desired.

Very good catch - I've updated and added the regex, thanks.

> How is this working without libshared? The release tarball has it included?

This is using the `release` tarball, which has the `libshared` source included.

ryandesign commented on office/timewarrior/Portfile:
```diff
@@ -0,0 +1,46 @@
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.0
```
Usually we want to use the latest version of a portgroup, which here is cmake 1.1. If you change that, then the line `cmake.out_of_source yes` can be deleted, because that is the default of cmake 1.1.

herbygillot replied:
Good point; fixed.
